crates/astria-sequencer/src/app.rs

```diff
@@ -2306,6 +2311,149 @@ mod test {
         assert_eq!(deposits[0], expected_deposit);
     }
 
+    // it's a test, so allow a lot of lines
+    #[tokio::test]
+    #[allow(clippy::too_many_lines)]
+    async fn app_execution_results_match_proposal_vs_after_proposal() {
```

Comment:
Should we have a test, perhaps using an snapshot to validate that when we only call finalize block on something (with actual transactions) that we reach the same result as what we generate by doing prepare/process/finalize?

Reply:
this test already tests that, it tests only finalize_block (full node), prepare_proposal, then process_proposal, then finalize_block (proposer node), and process_proposal then finalize_block (non proposer validator) and ensures the results are the same. and it uses both normal and bridge txs